### PR TITLE
Rework CLI class

### PR DESCRIPTION
**Pull request by nforro**
Allow supplying sets of tools to `pkgcomparetool` option
Do not include csmock into default set of pkgcomparetools
Move argument checks to `CLI`
Reword help messages to make them consistent and more clear

Resolves: #220

### CODE REVIEW

File: rebasehelper/cli.py [hunk 1]
```diff
@@ -54,112 +87,121 @@ def add_args(self):
...
         )
         self.parser.add_argument(
             "--pkgcomparetool",
-            default=False,
-            help="Select the tool for comparing two packages [pkgdiff, rpmdiff, abipkgdiff, csmock]. All compare tools"
-                 " are run by default."
+            # FIXME: set of choices could be obtained dynamically
+            choices={"pkgdiff", "rpmdiff", "abipkgdiff", "csmock"},
+            default={"pkgdiff", "rpmdiff", "abipkgdiff"},
+            type=lambda s: set(s.split(',')),
```
Comment (thozza) [inline rebasehelper/cli.py]: I was thinking more about using action 'append' and allow user to use the argument more than once, but I'm OK also with your solution.

**Comment by thozza** [comment]
@nforro LGTM. BTW you could use ```checkers_runner.get_supported_tools()``` and similar to get the list of supported tools dynamically. There should be an equivalent for each of the *tools.

**Comment by nforro** [comment]
@thozza: Implemented, including getting default choices.

File: rebasehelper/build_helper.py [hunk 2]
```diff
@@ -671,3 +672,9 @@ def get_supported_tools(cls):
         :return: list of supported build tools
         """
         return build_tools.keys()
+
+    @classmethod
+    def get_default_tool(cls):
+        """Returns default build tool"""
+        default = [k for k, v in six.iteritems(build_tools) if hasattr(v, 'DEFAULT') and v.DEFAULT]
```
Comment (thozza) [inline rebasehelper/build_helper.py]: @nforro you can add the DEFAULT attribute to the BuildToolBase class and set it to ```False```. Then only in the default tool class definition you can set it to ```True```. This way you won't have to use hasattr and the property of the class will be then explicitly set every time.

File: rebasehelper/checker.py [hunk 3]
```diff
@@ -129,6 +129,10 @@ def get_supported_tools(self):
         """Return list of supported tools"""
         return self.plugin_classes.keys()
 
+    def get_default_tools(self):
+        """Return list of default tools"""
+        return [k for k, v in six.iteritems(self.plugin_classes) if hasattr(v, 'DEFAULT') and v.DEFAULT]
```
Comment (thozza) [inline rebasehelper/checker.py]: The same as with build tools.

**Comment by thozza** [comment]
@nforro I had two comments to your new changes. Otherwise the changes look good!

**Comment by nforro** [comment]
@thozza: Thanks. I made the suggested changes.

**Comment by thozza** [comment]
@nforro looks good!

**Pull request merged**